stl/inc/xspan

```diff
@@ -0,0 +1,238 @@
+
+#pragma once
+#ifndef _XSPAN_
+#define _XSPAN_
+#include <yvals_core.h>
+#if _STL_COMPILER_PREPROCESSOR
+#include <array> // array
+#include <cstddef> // ptrdiff_t
+#include <type_traits> // remove_cv
+#include <xmemory> // _Compressed_pair, _One_then_variadic_args_t, _Zero_then_variadic_args_t
+
+#pragma pack(push, _CRT_PACKING)
+#pragma warning(push, _STL_WARNING_LEVEL)
+#pragma warning(disable : _STL_DISABLED_WARNINGS)
+_STL_DISABLE_CLANG_WARNINGS
+#pragma push_macro("new")
+#undef new
+
+_STD_BEGIN
+
+constexpr size_t dynamic_extent = numeric_limits<size_t>::max();
+
+namespace details {
+    template <size_t _Extent>
+    class _Extent_type {
+    public:
+        using index_type = size_t;
+
+        constexpr _Extent_type() noexcept = default;
+
+        template <index_type _OtherExtent>
+        constexpr _Extent_type(_Extent_type<_OtherExtent> _Ext) noexcept {
+            static_assert(_OtherExtent == _Extent || _OtherExtent == dynamic_extent,
+                "Mismatch between static extent and size of initializing data.");
+#if _CONTAINER_DEBUG_LEVEL != 0
+            _STL_VERIFY(_Ext.size() == _Extent, "Mismatch between static extent and size of initializing data");
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+        }
+
+        constexpr _Extent_type([[maybe_unused]] index_type _Size) noexcept {
+#if _CONTAINER_DEBUG_LEVEL != 0
+            _STL_VERIFY(_Size == _Extent, "Mismatch between dynamic extent and size of initializing data");
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+        }
+
+        constexpr index_type size() const noexcept {
+            return _Extent;
+        }
+    };
+
+    template <>
+    class _Extent_type<dynamic_extent> {
+    public:
+        using index_type = size_t;
+
+        constexpr _Extent_type() noexcept : _Mysize(index_type(0)) {}
+
+        template <index_type _OtherExtent>
+        explicit constexpr _Extent_type(_Extent_type<_OtherExtent> _Ext) noexcept : _Mysize(_Ext.size()) {}
+
+        explicit constexpr _Extent_type(index_type _Size) noexcept : _Mysize(_Size) {
+#if _CONTAINER_DEBUG_LEVEL > 0
+            _STL_VERIFY(_Mysize >= 0, "Mismatch between dynamic extent and size of initializing data");
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+        }
+
+        constexpr index_type size() const noexcept {
+            return _Mysize;
+        }
+
+    private:
+        index_type _Mysize;
+    };
+} // namespace details
+
+// [views.span], class template span
+template <class _Ty, size_t _Extent = dynamic_extent>
+class span {
+public:
+    using element_type    = _Ty;
+    using value_type      = remove_cv_t<_Ty>;
+    using index_type      = size_t;
+    using difference_type = ptrdiff_t;
+    using pointer         = _Ty*;
+    using const_pointer   = const _Ty*;
+    using reference       = _Ty&;
+    using const_reference = const _Ty&;
+    // using iterator               = _Span_iterator<pointer>;
+    // using const_iterator         = _Span_const_iterator<const_pointer>;

```

Comment:
Design: Should `span<const T>` have an `iterator` and `const_iterator` that are physically the same type?

Reply:
Locally I have only a `_Span_iterator` now